### PR TITLE
Java: Introduce LoC summary metric query

### PR DESCRIPTION
**Pull request by tamasvajk**

### CODE REVIEW

File: java/ql/src/Metrics/Summaries/LinesOfCode.ql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * @description The total number of lines of code across all files. This is a useful metric of the size of a database.
+ *              For all files that were seen during the build, this query counts the lines of code, excluding whitespace
+ *              or comments.
+ * @kind metric
+ * @tags summary
+ */
+
+import java
+
+select sum(File f, int i | i = f.getNumberOfLinesOfCode() | i)
```
Comment (aschackmull) [inline java/ql/src/Metrics/Summaries/LinesOfCode.ql:13]: Assuming that `getNumberOfLinesOfCode` is functional, this simplifies to
```suggestion
select sum(File f | f.getNumberOfLinesOfCode())
```

Comment (tamasvajk) [reply java/ql/src/Metrics/Summaries/LinesOfCode.ql:13]: I think I tried this, and I got an error along the lines of "sum can only be applied to int or float".

Comment (aschackmull) [reply java/ql/src/Metrics/Summaries/LinesOfCode.ql:13]: Ah, I guess you'd need an extra `|`, then:
```suggestion
select sum(File f | | f.getNumberOfLinesOfCode())
```

Comment (yo-h) [reply java/ql/src/Metrics/Summaries/LinesOfCode.ql:13]: We probably want to exclude counting "code" in XML files:
```suggestion
select sum(CompilationUnit f | f.fromSource() | f.getNumberOfLinesOfCode())
```

**Review by aschackmull — APPROVED** [review]

**Pull request merged**